compiler/transforms/accfg_config_overlap.py

```diff
@@ -84,6 +85,130 @@ def match_and_rewrite(self, op: accfg.SetupOp, rewriter: PatternRewriter, /):
         )
 
 
+class LoopLevelSetupAwaitOverlapPattern(RewritePattern):
+    """
+    Converts an `scf.for` loop to have setup/launch overlaps
+
+    Given the following IR:
+    ```
+     %s0 = setup ... // loop-initial state
+     scf.for (%i = %lb to %ub step %step) iter_args(%l0 = %s0) ... {
+       %l1 = setup from %l0 to ("i" = %i)
+       %t = launch(%l1)
+       await(%t)
+       yield %l1
+     }
+    ```
+
+    1. We grab the first setup op inside the loop, with all dependencies
+    2. We insert a copy of the setup op before the loop, replacing dependencies with the loop inputs (%lb)
+    3. We insert a copy of the setup af the end of the loop, replacing dependencies with the next iterations variables
+       (which we get from the yield / by adding %step to %i)
+    4. We erase the original setup op.
+
+
+    This results in the following IR post-optimization:
+
+    ```
+     %s0 = setup ... // loop-initial state
+     %l1 = setup from %l0 to ("i" = %lb)
```

Comment:
This seems incorrect
```suggestion
     %l1 = setup from %s0 to ("i" = %lb)
```